pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -2243,6 +2243,176 @@ def is_hermitian(self):
+    integer :math:`L`, flip a target qubit if :math:`\sigma \geq L`. Alternatively, the flipping condition can
+    be :math:`\sigma < L`.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 0
+    * Gradient recipe: None
+
+    Args:
+        value (int): The value :math:`L` that the state's decimal representation is compared against.
+        geq (bool): If set to `True`, the comparison made will be :math:`\sigma \geq L`. If `False`, the comparison
+            made will be :math:`\sigma < L`.
+        control_wires (Union[Wires, Sequence[int], or int]): Deprecated way to indicate the control wires.
+            Now users should use "wires" to indicate both the control wires and the target wire.
+        wires (Union[Wires, Sequence[int], or int]): control wire(s) followed by a single target wire where
+            the operation acts on.
+    """
+    is_self_inverse = True
+    num_wires = AnyWires
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    grad_method = None
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        value,
+        geq=True,
+        control_wires=None,
+        wires=None,
+        do_queue=True,
+    ):
+
+        if value is None:
+            raise ValueError("Must specify the integer value to compare against.")
+        if not isinstance(value, int):
+            raise ValueError("The comparable value must be an integer.")
+        if wires is None:
+            raise ValueError("Must specify the target wire where the operation acts on.")
+        if control_wires is None:
+            if len(wires) > 1:
+                control_wires = Wires(wires[:-1])
+                wires = Wires(wires[-1])
+            else:
+                raise ValueError(
+                    "IntegerComparator: wrong number of wires. "
+                    f"{len(wires)} wire(s) given. Need at least 2."
+                )
+        else:
+            wires = Wires(wires)
+            control_wires = Wires(control_wires)
+
+            warnings.warn(
+                "The control_wires keyword will be removed soon. "
+                "Use wires = (control_wires, target_wire) instead. "
+                "See the documentation for more information.",
+                category=UserWarning,
+            )
+
+            if len(wires) != 1:
+                raise ValueError("IntegerComparator accepts a single target wire.")
+
+        total_wires = control_wires + wires
+
+        self.hyperparameters["control_wires"] = control_wires
+        self.hyperparameters["value"] = value
+        self.hyperparameters["geq"] = geq
+        self.geq = geq
+        self.value = value
+
+        super().__init__(wires=total_wires, do_queue=do_queue)
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        return base_label or "X"
```

Comment:
```suggestion
        return base_label or f">{self.value}" if self.geq else f"<{self.value}"
```
Not sure if I have the directions right.

Reply:
Yep that's right! Small correct for the `>` case, added `=`